aider/main.test.ts

```diff
@@ -0,0 +1,184 @@
+import { describe, expect, it } from "bun:test";
+import {
+  execContainer,
+  findResourceInstance,
+  runContainer,
+  runTerraformApply,
+  runTerraformInit,
+  testRequiredVariables,
+  type TerraformState,
+} from "../test";
+
+// Executes the coder script after installing bash (required for the script)
+const executeScriptInContainerWithBash = async (
+  state: TerraformState,
+  image = "alpine",
+  extraCommands = "",
+): Promise<{
+  exitCode: number;
+  stdout: string[];
+  stderr: string[];
+}> => {
+  const instance = findResourceInstance(state, "coder_script");
+  const id = await runContainer(image);
+
+  // Install bash and set up the minimal environment needed
+  await execContainer(id, [
+    "sh",
+    "-c",
+    `
+    apk add --no-cache bash
+    mkdir -p /home/coder/bin
+    touch /home/coder/.aider.log
+    ${extraCommands}
+  `,
+  ]);
+
+  // Run the script with the environment variables from extraCommands
+  // Modifying to preserve environment variables
+  const resp = await execContainer(id, [
+    "bash",
+    "-c",
+    `
+    # Source any environment variables that might have been set
+    if [ -f /tmp/env_vars.sh ]; then
+      source /tmp/env_vars.sh
+    fi
+    
+    # Run the script
+    ${instance.script}
+  `,
+  ]);
+  const stdout = resp.stdout.trim().split("\n");
+  const stderr = resp.stderr.trim().split("\n");
+  return {
+    exitCode: resp.exitCode,
+    stdout,
+    stderr,
+  };
+};
+
+describe("aider", async () => {
+  await runTerraformInit(import.meta.dir);
+
+  testRequiredVariables(import.meta.dir, {
+    agent_id: "foo",
+  });
+
+  it("installs aider with default settings", async () => {
+    const state = await runTerraformApply(import.meta.dir, {
+      agent_id: "foo",
+    });
+
+    // Install bash and run the script
+    const output = await executeScriptInContainerWithBash(state);
+
+    // Verify that the script at least attempted to set up Aider
+    expect(output.stdout).toContain("Setting up Aider AI pair programming...");
+  });
+
+  it("uses tmux when specified", async () => {
+    const state = await runTerraformApply(import.meta.dir, {
+      agent_id: "foo",
+      use_tmux: true,
+      use_screen: false,
+    });
+
+    // Instead of running the script, just verify the script content
+    // to ensure the tmux parameter is being properly applied
+    const instance = findResourceInstance(state, "coder_script");
+
+    // Check for the correct tmux condition with the interpolated value
+    expect(instance.script.includes('if [ "true" = "true" ]')).toBe(true);
+
+    // Also check for a unique string only present when tmux is used in the script
+    expect(instance.script.includes("tmux new-session -d -s")).toBe(true);
```

Comment:
This check will always pass - that string will be present in the script regardless of the variable values. It'd be more useful to check stdout.